### PR TITLE
FxCop -> Net Analyzers (part 2)

### PR DESCRIPTION
**Pull request by Bond-009**

### CODE REVIEW

**Review by barronpm — APPROVED** [review]

File: Jellyfin.Data/DayOfWeekHelper.cs [hunk 1]
```diff
@@ -1,67 +1,21 @@
...
-            if (day == DynamicDayOfWeek.Saturday
-                || day == DynamicDayOfWeek.Weekend
-                || day == DynamicDayOfWeek.Everyday)
-            {
-                days.Add(DayOfWeek.Saturday);
-            }
-
-            return days;
+                DynamicDayOfWeek.Everyday => new[] { DayOfWeek.Sunday, DayOfWeek.Monday, DayOfWeek.Tuesday, DayOfWeek.Wednesday, DayOfWeek.Thursday, DayOfWeek.Friday, DayOfWeek.Saturday, DayOfWeek.Sunday },
+                DynamicDayOfWeek.Weekday => new[] { DayOfWeek.Sunday, DayOfWeek.Monday, DayOfWeek.Tuesday, DayOfWeek.Wednesday, DayOfWeek.Thursday, DayOfWeek.Friday },
```
Comment (crobibero) [inline Jellyfin.Data/DayOfWeekHelper.cs:15]:
```suggestion
                DynamicDayOfWeek.Weekday => new[] { DayOfWeek.Monday, DayOfWeek.Tuesday, DayOfWeek.Wednesday, DayOfWeek.Thursday, DayOfWeek.Friday },
```

**Review by crobibero — CHANGES_REQUESTED** [review]

**Review by crobibero — APPROVED** [review]

File: Jellyfin.Data/DayOfWeekHelper.cs [hunk 2]
```diff
@@ -1,67 +1,21 @@
...
-
-            if (day == DynamicDayOfWeek.Saturday
-                || day == DynamicDayOfWeek.Weekend
-                || day == DynamicDayOfWeek.Everyday)
-            {
-                days.Add(DayOfWeek.Saturday);
-            }
-
-            return days;
+                DynamicDayOfWeek.Everyday => new[] { DayOfWeek.Sunday, DayOfWeek.Monday, DayOfWeek.Tuesday, DayOfWeek.Wednesday, DayOfWeek.Thursday, DayOfWeek.Friday, DayOfWeek.Saturday, DayOfWeek.Sunday },
```
Comment (h1dden-da3m0n) [inline Jellyfin.Data/DayOfWeekHelper.cs:14]: was Sunday placed in this statement twice on purpose? (once before Monday and once on the end after Saturday)

Comment (h1dden-da3m0n) [reply Jellyfin.Data/DayOfWeekHelper.cs:14]: I see, and yeah I noticed this in some other context but failed to connect the dots. Thank you for the clarification :wink:

File: MediaBrowser.Common/Json/JsonDefaults.cs [hunk 3]
```diff
@@ -61,7 +61,7 @@ public static class JsonDefaults
         /// If the defaults must be modified the author must use the copy constructor.
         /// </remarks>
         /// <returns>The default <see cref="JsonSerializerOptions" /> options.</returns>
-        public static JsonSerializerOptions GetOptions()
+        public static JsonSerializerOptions Options
```
[the remark this answers is not in the document]
Comment (Bond-009) [reply MediaBrowser.Common/Json/JsonDefaults.cs:64]: yep

**Review by crobibero — APPROVED** [review]

File: Jellyfin.Data/DayOfWeekHelper.cs [hunk 4]
```diff
@@ -12,8 +12,8 @@ public static DayOfWeek[] GetDaysOfWeek(DynamicDayOfWeek day)
             return day switch
             {
                 DynamicDayOfWeek.Everyday => new[] { DayOfWeek.Sunday, DayOfWeek.Monday, DayOfWeek.Tuesday, DayOfWeek.Wednesday, DayOfWeek.Thursday, DayOfWeek.Friday, DayOfWeek.Saturday, DayOfWeek.Sunday },
-                DynamicDayOfWeek.Weekday => new[] { DayOfWeek.Sunday, DayOfWeek.Monday, DayOfWeek.Tuesday, DayOfWeek.Wednesday, DayOfWeek.Thursday, DayOfWeek.Friday },
```
Comment (MrTimscampi) [inline Jellyfin.Data/DayOfWeekHelper.cs]: Sunday being a weekend or not is very culture dependent. Not sure if it should be hardcoded or not.

**Pull request merged**